### PR TITLE
try importing esmpy (aliased to ESMF) for esmpy >= 8.4.0 fixes #212

### LINKED ISSUE #212: ESMpy 8.4.0 import changes
`ESMpy` 8.4.0, which got on conda-forge 45 min ago, changes how it needs to be imported.

```
# Before
import ESMF

# After
import esmpy
```

A quick fix is to pin `esmpy < 8.4` in the environments.

### PR DESCRIPTION
**Pull request by raphaeldussin**
fall back on old name if the import fails

### CODE REVIEW

File: xesmf/backend.py [hunk 1]
```diff
@@ -18,7 +18,10 @@
 import os
 import warnings
 
-import ESMF
+try:
+    import esmpy as ESMF
+except:
```
Comment (aulemahal) [inline xesmf/backend.py:23]:
```suggestion
except ImportError:
```
Should work the same and satisfy the linter.

Comment (raphaeldussin) [reply xesmf/backend.py:23]: thanks! will try it out now

**Comment by raphaeldussin** [comment]
@aulemahal I'm getting the test failing for the spatialAverager with `AttributeError: 'CoordinateSequence' object has no attribute 'array_interface'`

can you take a look?

**Comment by aulemahal** [comment]
That's shapely 2.0... ugh...

Pinning it to `< 2.0` should work for now. I'll look into it and try to push a fix.

**Comment by raphaeldussin** [comment]
ok I'll put a pin for now before we get flooded with broken installs with the new esmpy

**Comment by raphaeldussin** [comment]
fixes #212 

this is ready for review (then squash and merge) and somewhat time sensitive.

**Review by huard — APPROVED** [review]

**Comment by huard** [comment]
Just add a note to CHANGES about this.

File: ci/environment.yml [hunk 2]
```diff
@@ -14,6 +14,6 @@ dependencies:
   - pydap
   - pytest
   - pytest-cov
-  - shapely
+  - shapely<2.0
```
Comment (zmoon) [inline ci/environment.yml:17]: Should this have been done in [the cf feedstock](https://github.com/conda-forge/xesmf-feedstock/blob/main/recipe/meta.yaml) as well?

**Pull request merged**